ragstack-e2e-tests/e2e_tests/test_utils/vector_store_handler.py

```diff
@@ -51,18 +52,22 @@ def new_llamaindex_vector_store(self, **kwargs) -> EnhancedLlamaIndexVectorStore
 class VectorStoreHandler(ABC):
     def __init__(
         self,
-        implementation: VectorStoreImplementation,
         supported_implementations: List[VectorStoreImplementation],
     ):
-        self.implementation = implementation
+        self.implementation = VectorStoreImplementation.UNINITIALIZED
```

Comment:
I'd say this is the downside with this refactoring, but I think it suggests that we should have a separate implementation for the astra client and cassio

Reply:
not really, the fact is that AstraDB works with using astrapy both in cassandra and astradb mode

Reply:
Yes, but it remains that cassandra mode requires some additional setup. If the shared functionality remains in the `AstraDBVectorStoreHandler`, an `CassIOAstraDBVectorStoreHandler` (naming tbd) can handle that additional setup. 

That would let us do 
```
  if vector_database_type == "astradb":
        return AstraDBVectorStoreHandler()
   elif vector_database_type == "astradb-cassio":
      return CassIOAstraDBVectorStoreHandler()
    elif vector_database_type == "local-cassandra":
        return CassandraVectorStoreHandler()
```

and then test fixtures will not need to specify `Implementation`. 

```
@pytest.fixture
def astra_db():
    handler = get_vector_store_handler()
    context = handler.before_test()
    yield context
    handler.after_test()
```

It then splits up the e2e tests in CI between cassandra and data API modes, which I see as beneficial. 

But, if you disagree, I can work on fixing the singleton in another manner. This was a blocker on the `MyPy` work from last week, as it wanted some refactoring in this area in particular.